### PR TITLE
[improvement](compaction)Support segcompaction for inverted index

### PR DESCRIPTION
**Pull request by qidaye**
# Proposed changes

Since Doris supports segcompaction #12866 during loading, inverted index support is also needed.

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [x] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by qidaye** [comment]
run buildall

File: be/src/olap/rowset/beta_rowset_writer.cpp [hunk 1]
```diff
@@ -255,13 +279,35 @@ Status BetaRowsetWriter::_rename_compacted_segment_plain(uint64_t seg_id) {
...
         _clear_statistics_for_deleting_segments_unsafe(seg_id, seg_id);
     }
-    ++_num_segcompacted;
     ret = rename(src_seg_path.c_str(), dst_seg_path.c_str());
     if (ret) {
         LOG(WARNING) << "failed to rename " << src_seg_path << " to " << dst_seg_path
                      << ". ret:" << ret << " errno:" << errno;
         return Status::Error<ROWSET_RENAME_FILE_FAILED>();
     }
+    // rename remaining inverted index files
```
Comment (freemandealer) [inline be/src/olap/rowset/beta_rowset_writer.cpp:288]: the following code is similar to the code beginning with line 227 except for the path logic. Can we make a common function here to eliminate duplication?

File: be/src/olap/rowset/beta_rowset.cpp [hunk 2]
```diff
@@ -88,6 +88,22 @@ std::string BetaRowset::local_segment_path_segcompacted(const std::string& table
     return fmt::format("{}/{}_{}-{}.dat", tablet_path, rowset_id.to_string(), begin, end);
 }
 
+std::string BetaRowset::inverted_index_file_path(const string& rowset_dir,
+                                                 const RowsetId& rowset_id, int segment_id,
+                                                 int64_t index_id) {
+    // {rowset_dir}/{schema_hash}/{rowset_id}_{seg_num}_{index_id}.idx
+    return fmt::format("{}/{}_{}_{}.idx", rowset_dir, rowset_id.to_string(), segment_id, index_id);
+}
```
[the remark this answers is not in the document]
Comment (qidaye) [reply be/src/olap/rowset/beta_rowset.cpp]: done

File: be/src/olap/rowset/beta_rowset_writer.cpp (hunk 1 above)
Comment (qidaye) [reply be/src/olap/rowset/beta_rowset_writer.cpp:271]: done

**Comment by qidaye** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.1 seconds
 stream load tsv:          454 seconds loaded 74807831229 Bytes, about 157 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          73 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230320100759_clickbench_pr_117264.html

**Comment by freemandealer** [comment]
LGTM

**Review by freemandealer — DISMISSED** [review]
LGTM

**Comment by qidaye** [comment]
run buildall

**Comment by qidaye** [comment]
run buildall

**Comment by qidaye** [comment]
run buildall

**Review by airborne12 — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**